examples/CairoUI/CairoExampleUI.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * DISTRHO Plugin Framework (DPF)
+ * Copyright (C) 2012-2019 Filipe Coelho <falktx@falktx.com>
+ *
+ * Permission to use, copy, modify, and/or distribute this software for any purpose with
+ * or without fee is hereby granted, provided that the above copyright notice and this
+ * permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
+ * TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
+ * NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
+ * DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
+ * IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
+ * CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include "DistrhoUI.hpp"
+#include "DemoWidgetBanner.hpp"
+#include "DemoWidgetClickable.hpp"
+#include "Window.hpp"
+
+#include <memory>
+
+START_NAMESPACE_DISTRHO
+
+class CairoExampleUI : public UI
+{
+public:
+    CairoExampleUI()
+        : UI(200, 200)
+    {
+        DemoWidgetClickable* widgetClickable = new DemoWidgetClickable(this);
+        fWidgetClickable.reset(widgetClickable);
+        fWidgetClickable->setSize(50, 50);
+        fWidgetClickable->setAbsolutePos(100, 100);
+
+        DemoWidgetBanner* widget_banner = new DemoWidgetBanner(this);
+        fWidgetBanner.reset(widget_banner);
+        fWidgetBanner->setSize(180, 80);
+        fWidgetBanner->setAbsolutePos(10, 10);
+    }
+
+    ~CairoExampleUI()
+    {
+    }
+
+    void onDisplay()
+    {
+        const GraphicsContext& gc = getParentWindow().getGraphicsContext();
+        cairo_t* cr = gc.cairo;
+
+        cairo_set_source_rgb(cr, 1.0, 0.8, 0.5);
+        cairo_paint(cr);
+
+        cairo_set_source_rgb(cr, 0.0, 0.8, 0.5);
+
+        Point<int> tp1(50, 100);
+        Point<int> tp2 = tp1 + Point<int>(20, 40);
+        Point<int> tp3 = tp1 + Point<int>(-20, 40);
+
+        Triangle<int>(tp1, tp2, tp3).draw(&gc);
+
+        tp1 += Point<int>(0, 20);
+        tp2 += Point<int>(5, 20);
+        tp3 += Point<int>(-5, 20);
+
+        Triangle<int>(tp1, tp2, tp3).draw(&gc);
+
+        tp1 += Point<int>(0, 20);
+        tp2 += Point<int>(5, 20);
+        tp3 += Point<int>(-5, 20);
+
+        Triangle<int>(tp1, tp2, tp3).draw(&gc);
+
+        cairo_set_source_rgb(cr, 0.8, 0.5, 0.0);
+
+        tp1 += Point<int>(-5, 40);
+
+        Rectangle<int>(tp1, 10, 10).draw(&gc);
+    }
+
+    void parameterChanged(uint32_t index, float value)
+    {
+        // unused
+        (void)index;
+        (void)value;
+    }
+
+private:
+    std::unique_ptr<DemoWidgetClickable> fWidgetClickable;
```

Comment:
I am not sure if `std::unique_ptr` works for c++98, there is a DPF provided ScopedPointer for things like this.